src/azure-cli/azure/cli/command_modules/vm/custom.py

```diff
@@ -5322,124 +5258,97 @@ def _set_log_analytics_workspace_extension(cmd, resource_group_name, vm, vm_name
+    _user_mi = identity.pop('userAssignedIdentities', None)
+    if _user_mi:
+        identity.update({'user_assigned_identities': _user_mi})
 
-def remove_disk_encryption_set_identity(cmd, client, resource_group_name, disk_encryption_set_name,
-                                        mi_system_assigned=None, mi_user_assigned=None):
-    DiskEncryptionSetUpdate = cmd.get_models('DiskEncryptionSetUpdate', operation_group='disk_encryption_sets')
-    client = _compute_client_factory(cmd.cli_ctx)
+    user_identities_to_remove = []
+    if mi_user_assigned is not None:
+        existing_user_identities = {x.lower() for x in list(identity.get("user_assigned_identities", {}).keys())}
 
-    def getter(cmd, resource_group_name, disk_encryption_set_name):
-        return client.disk_encryption_sets.get(resource_group_name, disk_encryption_set_name)
+        # all user assigned identities will be removed if the length of mi_user_assigned is 0,
+        # otherwise the specified identity
+        user_identities_to_remove = {x.lower() for x in mi_user_assigned} \
+            if len(mi_user_assigned) > 0 else existing_user_identities
+        non_existing = user_identities_to_remove.difference(existing_user_identities)
+        if non_existing:
+            from azure.cli.core.azclierror import InvalidArgumentValueError
+            raise InvalidArgumentValueError("'{}' are not associated with '{}', please provide existing user managed "
+                                            "identities".format(','.join(non_existing), disk_encryption_set_name))
+        if not list(existing_user_identities - user_identities_to_remove):
+            identity.pop('user_assigned_identities', None)
+            if identity['type'] == "UserAssigned":
+                identity['type'] = "None"
+            elif identity['type'] == "SystemAssigned, UserAssigned":
+                identity['type'] = "SystemAssigned"
 
-    def setter(resource_group_name, disk_encryption_set_name, disk_encryption_set):
-        disk_encryption_set_update = DiskEncryptionSetUpdate(identity=disk_encryption_set.identity)
-        return client.disk_encryption_sets.begin_update(resource_group_name, disk_encryption_set_name,
-                                                        disk_encryption_set_update)
+    if remove_system_assigned_identity:
+        identity['type'] = "None" if identity['type'] == "SystemAssigned" else "UserAssigned"
 
-    return _remove_disk_encryption_set_identities(cmd, resource_group_name, disk_encryption_set_name,
-                                                  mi_system_assigned, mi_user_assigned, getter, setter)
+    if user_identities_to_remove:
+        if identity['type'] not in ["None", "SystemAssigned"]:
+            for _id in user_identities_to_remove:
+                identity['user_assigned_identities'].pop(_id, None)
+    args['identity'] = identity
+    result = LongRunningOperation(cmd.cli_ctx)(_Update(cli_ctx=cmd.cli_ctx)(command_args=args))
```

Comment:
I think it should be _Create command instead of _Update commnd, since _Update command also contains GET operation which is unnecessary.